src/qml/modules/Shotcut/Controls/GradientControl.qml

```diff
@@ -47,7 +47,7 @@ RowLayout {
                stopIndex == colorList.length - 1 ? parent.width - width :
                stopIndex * (parent.width / (colorList.length - 1)) - (width / 2)
             y: 0
-            color: colorList[stopIndex]
+            color: colorList[stopIndex] != undefined ? colorList[stopIndex] : "gray"
```

Comment:
`undefined` is not a keyword in JavaScript, and maybe it could be defined as a global variable somewhere to something with a type. Moreover, this is not using a strict comparison operator `!==`. This should be 
`typeof colorList[stopIndex] !== 'undefined' ? colorList[stopIndex] : "gray"`

You can search our code to see where I have used it previously.

Reply:
done